internal/deploy/docker.go

```diff
@@ -92,13 +112,61 @@ func (c *dockerDeploymentManifest) Inspect(service string) (*ServiceManifest, er
 		ID:         inspect.ID,
 		Name:       strings.TrimPrefix(inspect.Name, "/"),
 		Connection: service,
-		Hostname:   inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Alias:      inspect.NetworkSettings.Networks["fleet_default"].Aliases[0],
+		Hostname:   inspect.Config.Hostname,
+		Platform:   inspect.Platform,
 	}, nil
 }
 
+// Mount will mount a service with ability to perform actions within that services environment
+// TODO: Not a fan of passing in installType here, should think about abstracting that portion out more
+func (c *dockerDeploymentManifest) Mount(service string, installType string) (installer.Package, error) {
+	log.WithFields(log.Fields{
+		"service":     service,
+		"installType": installType,
+	}).Trace("Mounting service for configuration")
+
+	container, _ := c.Inspect(service)
+	var install installer.Package
+	if strings.EqualFold(service, "elastic-agent") {
+		switch installType {
+		case "tar":
+			install = installer.NewElasticAgentTARPackage(container.Name, c.ExecIn, c.AddFiles)
```

Comment:
Not sure about this method, as we are coupling the operation of mounting things to services, with the Fleet test suite. Maybe the name should express that (i.e. _MountElasticAgentInstaller_) to avoid contributors in other suites (metricbeat) to use it.

Reply:
Yea I agree with the name change to make it more obvious